astropy/coordinates/angles/core.py

```diff
@@ -313,15 +314,14 @@ def to_string(
                 f"With decimal=True, separator cannot be used (got {sep=!r})"
             )
 
+        unit = self._convert_unit_to_angle_unit(unit)
         if unit is None:
             if sep == "dms":
                 unit = u.degree
             elif sep == "hms":
                 unit = u.hourangle
             else:
                 unit = self.unit
-        else:
```

Comment:
I think this one goes the wrong way - you're making the default case of `unit is None` a (little) slower, so I'd just remove the `u.Unit()` on the original line.

Reply:
OK